src/lib/es2015.core.d.ts

```diff
@@ -10,9 +10,7 @@ interface Array<T> {
      * @param thisArg If provided, it will be used as the this value for each invocation of
      * predicate. If it is not provided, undefined is used instead.
      */
-    find(predicate: (this: void, value: T, index: number, obj: Array<T>) => boolean): T | undefined;
-    find(predicate: (this: void, value: T, index: number, obj: Array<T>) => boolean, thisArg: undefined): T | undefined;
-    find<Z>(predicate: (this: Z, value: T, index: number, obj: Array<T>) => boolean, thisArg: Z): T | undefined;
+    find(predicate: (this: void, value: T, index: number, obj: Array<T>) => boolean, thisArg?: any): T | undefined;

```

Comment:
Do we need the `this: void`? It's not actually accurate because if a `thisArg` is passed, then `this` won't be `void`.
If someone really does want to pass `thisArg`, `this: void` would stop them, but if `this: void` is ommitted, they will correctly get an error in `--noImplicitThis` mode, and can work around it by explicitly annotating a `this` type for their callback.
```ts
declare function f(cb: (this: void) => void, thisArg?: any): void;
declare function g(cb: () => void, thisArg?: any): void;

// Can't get a correct type for `this`, even with explicit annotation
f(function(this: string) {
    this.toLowerCase();
}, "");

// Works
g(function(this: string) {
    this.toLowerCase();
}, "");
```

Reply:
thanks. should be fixed now.